### DISCUSSION TITLE
Some problems on the features introduced by the dev branch

### DISCUSSION THREAD

**Discussion by bernardstef** [philibertc/micronova_controller#5] [General]
I followed the instructions, and I can retrieve data in Node red through MQTT...
Just wanted to say thanks !!

Edit : I can in fact see that only ambient temp is right. Others values are off
The adresses doesn't seem to fit my stove (Freepoint Sharp with Micronova Agua IOT , model : W001 ) I'll try to find the right addresses from the serial RX if i'm able to. It doesn't respond to on/off and flame power commands too.

**Comment by philibertc** [comment]
Hello,
Strange, all the addresses I use are the same ones sent by the Agua IOT platform.
Can you try this: https://raw.githubusercontent.com/philibertc/micronova_controller/master/tests/esp8266/test1_on_off.ino please ?
This test turns on then 2 sec later off the stove, no MQTT here, just a simple test.

**Reply by bernardstef** [reply]
Thanks for the reply, your code is working, it turns on then off.

I uploaded your other test code (read) and connected to serial from the ESP, I can see this (if it helps) :

[https://imgur.com/a/7muY4ZH](https://imgur.com/a/7muY4ZH)

Using another serial adapter, on the stove side with a serial monitor software, I can't find something I can understand (I can't spot the ambient temp value, this one is working right)

[https://imgur.com/a/nSNSkzq](https://imgur.com/a/nSNSkzq)

Again, I have a brand new stove, board is W001 from Micronova, maybe they changed some address.

**Reply by bernardstef** [reply]
Oh, wait a moment.... I'm asking myself if the "easy connect plus " BT and WIFI option I have on my stove is messing with the serial communication.. Trying right now..

**Reply by philibertc** [reply]
Happy to hear the test is working.
The read test seams also okay.

**Reply by bernardstef** [reply]
Unfortunately, even with the Micronova wifi disconnected, some values are off.. Fume temp goes from 34, 50.. with stove running.
It seem to change when the blower speed changes.

ON / OFF stove status is right (as well as the command to turn it on/off, thanks !), also, as well as room temp.
Maybe I'll find the right addresses ! ;)

**Reply by philibertc** [reply]
I have to same behaviour with my two stoves (2013 and 2018), the fumes temp is 0 when off and 0/50 when starting. I'm waiting for winter to do some tests with the stove fully started.

**Reply by philibertc** [reply]
I'm going to put a "known issues" paragraph in the README.

**Reply by bernardstef** [reply]
Thanks a lot, I'll be following this, tell me if I can be any help for some tests.
[https://imgur.com/a/Q072jMq](https://imgur.com/a/Q072jMq)

**Reply by philibertc** [reply]
Ooops, your imgur link is not valid.

**Reply by bernardstef** [reply]
Sorry ! Fixed. I'll make the serial port available on the back for convenience and later tests.

**Reply by philibertc** [reply]
Hello @bernardstef 
I found out why it was not responding to ON or OFF commands. I had a test in the program that I forgot to remove.
On my side, everything works normally now (except for the temperature of the smoke).

**Reply by bernardstef** [reply]
Hi !

Great, I tested and it works !
What is working : 
-Ambient temp,
-ON/OFF and ON / OFF Status
-Status (work, pellets loading, etc..),

I have some doubts with:
-Flame power data seems to go over 5 ? I've seen it at 6 and more than that, earlier but I don't remember how much. My stove setting goes up to 5.
- flame power command, I'm not sure, doesn't do much on mine (stays at what i've set on the panel)
- fume temp

Again, it's awesome, also the schematic is brilliant. Excellent !! :+1: 


You might be interested, I used another clever addition to my stove which is the pellet level sensor. Connected to an ESP8266 with node red. https://www.magicmanu.com/arduino/2020/12/alerte-niveau-reservoir-pour-poele-a-pellet/

**Reply by philibertc** [reply]
I will soon add the possibility to connect a ping sensor.
Normally, if you send more than 5 in mqtt, nothing should happen. I will test soon if I can change the power. But the reading works?

**Reply by bernardstef** [reply]
Great ! 
I just noticed that if you send a "On" command while the stove is fully started ("work" status) it starts again (pellet loading, etc..). Interesting behavior of the micronova board.

The reading of the flame power works, I just tested, it goes up to 16 when I set 5 on the display and wait for the flame to set.

Setting the flame to 1-5 via MQTT is a bit strange. For example, the flame is set on 5 and the graph (bars on the display) shows 5.
I send the command to set it to 1. Fan slows down, graph shows 1. But the setting on the menu is still 5. After a few moment, fan speeds up and the graph shows full bars.

To me it seems that after having sent the packet, it lure the stove into thinking that the actual flame is 1, and it need to get to 5. It doesn't change the setting (it might be another address in the memory?)
That's my experience! ;)

**Reply by philibertc** [reply]
Does the fume temperature reading works when fully started ?
I just found out that maybe the flame power should be set in the EEPROM. If it is the case, I will not implement it because if there is a bug (eg a MQTT message sent many times) it can break the EEPROM (because of limited number of writes).
EDIT: I will put a condition in the program not to restart the stove when it is already started.

**Reply by bernardstef** [reply]
Unfortunately, no, fume temp values are 0, 34 and 50 .. Indeed, they change when the stove is started but oscillating between those 3 values.
Edit: 105 shows up from time to time, although it's certainly not important.

Excellent !! that's awesome ..... !!

**Reply by philibertc** [reply]
Hmmm, this problem really bothers me.

**Reply by bernardstef** [reply]
How did you get those memory address ? , tell me if I can be any help to get this sorted.
EDIT  : Sorry I haven't see ! It was in the readme .. ;)
I was planing on using a temp sensor wired up to my other esp8266 initially, for the fume temp. Obviously reading it via serial is better.

My stove came with a wifi module "easy connect plus", on the phone app, via bluetooth of wifi (both options available) I can set the flame power and read the fume temp on my phone. It's connected via serial interface, probably the same (but on another socket on the main board)as we connect to (as i could see the TX led blinking very fast when I connected a ttl/usb converter to the serial, for tests the other day. Unplugged the wifi and led as blinking only when your module was sending/reading)

So those packets are traveling via the serial interface..

Also, I unplugged this wifi module so it doesn't interfere during tryouts.

**Reply by philibertc** [reply]
Hello,
I created a program that just reads the smoke temperature and sends it via the serial port. You can experiment with this. [program](https://raw.githubusercontent.com/philibertc/micronova_controller/master/tests/esp8266/test3_fumetemp.ino)

**Reply by bernardstef** [reply]
Hi,

Unfortunately, it's about the same. Values are not close to fume temp as I can see in the display panel ("info" menu) Values seems to change when I enter the menu on the stove control panel :

10:47:51.909 -> Param=5a value=3250 (outside the menu)
10:47:54.029 -> Param=5a value=00 (outside the menu)
10:47:56.109 -> Param=5a value=67103 (in the menu)

Fume temp was around 115 - 120°c during this test.

I installed Seraminet but can't seem to get anything other than 0xffff ... on the memory reading. I might have to use the optocouplers to get TX/RX on a single wire.

**Reply by philibertc** [reply]
3250 and 67103?? Normally the stove can sen no more than 255.
For the moment I suggest you use the program as is, until the problem is solved. I will notify you if I get an improvement.

**Reply by bernardstef** [reply]
Well I don't know but it showed up on the Arduino serial console
[https://imgur.com/a/jYfFLqm](https://imgur.com/a/jYfFLqm)
I reproduced it (entered the menu)

Maybe this evening I'll have time to try seraminet ! So we could get this sorted out ;)

**Reply by bernardstef** [reply]
I haven't had time to test yesterday, but when I get home I think I'll test this.
I don't know how to interpret the data in the hex array... Maybe convert to base 10 ?

https://stufapellet.forumcommunity.net/?t=53061907&st=45#entry401594179

**Reply by philibertc** [reply]
You can use an hex to decimal converter (yes, base 10).
What are you going to test ? I didn't really understand what was said in the forum comment.

**Reply by bernardstef** [reply]
Well, I'm currently tying to get Seraminet to get all the ram variables
but I can't get anything besides 0xfff ... I'm trying right now but nothing..
[https://imgur.com/a/RiOZ16Y](https://imgur.com/a/RiOZ16Y)

**Reply by philibertc** [reply]
What do you use to communicate with the stove? The ESP circuit?
If yes, what is the program you make to do so? I'm interested.

**Reply by bernardstef** [reply]
No, I just used some UART usb serial adapter, connected instead of the ESP, so I can use the opto isolator circuit. But it doesn't realy work right now.. (1200 , 8 bits, no parity 2 stops)

**Reply by bernardstef** [reply]
I tapped into the wifi module (covering the serial and mac address on the photo)
It's an esp32 wrover.
![image](https://user-images.githubusercontent.com/62932530/132387953-d94704dd-686f-4e45-9c1f-ac5ae0b7ac5e.png)


Something interesting, they seem to use MQTT too !
[https://pastebin.com/znfSf7xW](https://pastebin.com/znfSf7xW)

**Comment by philibertc** [comment]
Summary of the problems cited in this discussion:
- Nothing happens when ON or OFF is sent → Solved
- Flame power control does not work → Will be voluntarily unresolved (needs to be written on EEPROM)
- The flue gas temperature reading does not work → **Unresolved for everyone**

**Reply by bernardstef** [reply]
I change flame setting on the app from 2 to 1, I can see the esp transmetting this on the serial debug :
![image](https://user-images.githubusercontent.com/62932530/132389264-7f7b6ba7-c748-4151-b526-4da96488ef65.png)


![image](https://user-images.githubusercontent.com/62932530/132389443-718d9555-9693-4d53-be68-42851c250fce.png)

What do you thing about "get buffer reading" ? interresting too, might be corresponding to 0x01, 02, etc ?
Fume temp is fluctuating from 31 to 33, but I can't find it.

**Reply by philibertc** [reply]
Okay, it's EEPROM, so I will not implement it for security reasons.
Can you please test with the Eco Stop mode, the Ventilation, and what it sends to get the fume temp.
_________
C'est écrit sur l'EEPROM, je ne vais pas l'implémenter pour des raisons de sécurité.
Pouvez-vous tester en changeant les réglages de l'Eco Stop, de la ventilation. Il affiche dans les logs ce qu'il envoie pour obtenir la température des fumées ? EDIT: Je viens de voir ce que vous dites à propos de la température des fumées, mince !

**Reply by bernardstef** [reply]
Fan set from 1 to auto :

read for mask addr:007E  value:0001
C (1639379) [BASE_COM]: write in EEP address: 007E value:0


eco stop from on to off:

read for mask addr:001E  value:0001
write with mask addr:001E mask:FFFF value:0000

eco stop timer from 15 to 10 minutes:
read for mask addr:001B  value:000F
write with mask addr:001B mask:FFFF value:000A

thermostat setting from 17c to 14c:


read for mask addr:007D  value:0011
write with mask addr:007D mask:00FF value:000E

**Reply by bernardstef** [reply]
Fume temp was a steady 34c in this buffer reading, but I can't find it:

{"BufferId":1,"cmd":"GetBufferReading","Items":[32847,32878,32857,32844,32884,32839,33,32828,32842,32845,32894,32860,33020,32826,33019,32867,232,32831,32853,32855,32850,33021,32882,32827,32835,32833,32863,6,32870,32851,51,32841,2,33017,32866,32881,33016,32877,32846,32836,32858,32795,32823,32880,32830,32838,32837,32861,32864,32798,32829,32843,32848,32872,32875,33018,32871,32874,1,32832,32852,32883,32895,32873,32876,32862,52,32859,32893,32840,32869,32868,32849,32834,32854,32856,32879,32825,32865],"Values":[0,0,0,0,0,0,7,1,0,0,0,0,9,96,7,102,32,1,0,0,0,33,0,135,0,1,0,64,0,0,0,0,4,32,96,0,2,0,0,51,0,10,1,0,1,0,52,0,0,1,1,0,0,0,0,6,0,0,53,1,0,0,2,0,0,0,0,0,17,0,0,0,0,0,0,0,0,1,0]}}]

**Reply by philibertc** [reply]
Thanks! It's also EEPROM for the fans... But not for the eco stop and thermostat, I'm gonna implement that.

**Reply by philibertc** [reply]
In the information tab, the fume temp is displayed?

**Reply by bernardstef** [reply]
We're quite lucky I have this wifi module installed, I didn't order it ! The technician told me I can buy it for 200€... For me to find it was fitted in the stove, later ! :+1:

**Reply by bernardstef** [reply]
It is, indeed 

![image](https://user-images.githubusercontent.com/62932530/132391229-2ac46f17-0475-4a15-9cae-8f8e0a6e63eb.png)

**Reply by philibertc** [reply]
Analyzing your module will, I think, solve the problem.

**Reply by bernardstef** [reply]
Awesome ! !

**Reply by bernardstef** [reply]
Fume temp at 33°c if it can help

:"GetBufferReading","Items":[32847,32878,32857,32844,32884,32839,33,32828,32842,32845,32894,32860,33020,32826,33019,32867,232,32831,32853,32855,32850,33021,32882,32827,32835,32833,32863,6,32870,32851,51,32841,2,33017,32866,32881,33016,32877,32846,32836,32858,32795,32823,32880,32830,32838,32837,32861,32864,32798,32829,32843,32848,32872,32875,33018,32871,32874,1,32832,32852,32883,32895,32873,32876,32862,52,32859,32893,32840,32869,32868,32849,32834,32854,32856,32879,32825,32865],"Values":[0,0,0,0,0,0,7,1,0,0,0,0,9,96,7,102,32,1,0,0,0,33,0,135,0,1,0,64,0,0,0,0,3,32,96,0,2,0,0,51,0,10,1,0,1,0,52,0,0,1,1,0,0,0,0,22,0,0,53,1,0,0,2,0,0,0,0,0,17,0,0,0,0,0,0,0,0,1,0]}}]

**Reply by philibertc** [reply]
Hello,
can you change the flame power with that program: https://pastebin.com/UUwRkcJE
If it doesn't works, I recommend you to switch back to the program in the main branch (recently updated).

**Reply by bernardstef** [reply]
Hi!
I can't rest right now... I'll be able to, tomorrow evening I think!
I'll keep you posted

**Reply by philibertc** [reply]
Okay, thanks

**Reply by philibertc** [reply]
Awesome, fume temp reading works now!!
It is on 0x3E. I I have a micronova PO69 wifi module, which doesn't work anymore (impossible to connect to it), but it continuously sends commands to the stove. I was able to analyze it with a logic analyser.
____
Je possède un module wifi micronova PO69, qui ne fonctionne plus (impossible de se connecter dessus), mais il envoie en continu les commandes au poèle. J'ai donc pu l'analyser avec un logic analyser.

**Reply by bernardstef** [reply]
Excellent!!! 😁
Can't wait to get home and test!
I did search with the values I got from the serial console on the wifi module, but couldn't figure something out..
I'll get back to you this evening 👍

**Reply by bernardstef** [reply]
Hi,

Flame power on pastebin doesn't seem to make any change ... 

test2bis_read_all_addresses.ino outputs "'stoveRxData' was not declared in this scope" when compiling

---

Latest main stove controller.ino:
Fume temp: allright !! nice !!!

On/off command (sending N or F and checking by serial if ESP received it right) : doenst' want to work for me tonight, strange .. Might need some more tests? Used to work

Flame power:
command doesn't do something (it seems to messe with "Eco stop :on/off" on my app, shows weird values on this section on the app when I set some flame power via MQTT...). EDIT : I've seen strange number values instead of on/off, without sending flame power. Seems to be a bug on their side. 

Reading goes from 0 to 16, matching actual flame height or power I think, great !


Awesome work ..... :+1:

**Reply by philibertc** [reply]
Hello,
test2bis: I'm going to resolve that.
On/Off: You have to send ON or OFF.
Flame power: Sending 2,3,4,5 doesn't do anything now. 0 do the same thing than OFF and 1 = ON.
Flame power reading: strange, normally it's between 0 and 5???

**Reply by bernardstef** [reply]
Well, on mine it outputs 0-16 to MQTT (node red sends it directly to Domoticz for logging)

[https://imgur.com/a/FTtRxxA](https://imgur.com/a/FTtRxxA)

Noted for On/off ! Thanks I thought it changed but I guess I was thinking about something else !

**Reply by bernardstef** [reply]
Hi !

Thanks for the update with new values from the stove, but I have a little problem, my stove keeps turning on after the update to the latest code... Strange ! ...


https://photos.app.goo.gl/1yrjP1w6mnRMtoLw5

**Reply by bernardstef** [reply]
I confirmed, going back to the version "Prepare for new features" and no problem ... Really strange..

**Reply by philibertc** [reply]
Hello,
Strange, yes. I'm going to test what part of these new features cause the problem

**Reply by philibertc** [reply]
And for the flame power reading is it still between 0 and 16? If yes, when you read 6 for example what is displayed in the Easy Connect app?

**Reply by bernardstef** [reply]
Hum, I don't remember, i'll test again and tell you !

**Reply by bernardstef** [reply]
Flame power value (from 0 to 16) is  linked in some way with the app display ("real power" (from 0 to 5) section) , but the steps are different.
I tested two times going up or down so I'm confident with the different "steps" ;)

Fisrt digit is "real power" on the app. If I set "4" on the stove, "real power" stabilize to "4 after a moment.
1: 0 1 2 3
2: 4 5 6 7
3: 8 9 10 11
4: 12 13 14
5: 15 16

On the stove control panel, in the "info" then "flame" menu, the "real power" value is displayed (same as the app)

**Reply by philibertc** [reply]
<details>
  <summary><b>English</b></summary>
  
Okay, thanks.

Recap of the tests performed:
- read in RAM the value of the thermostat returns 0 (you can test it if you want by uncomment `getTempSet();` and `delay(100);` just before)
- tested by reading from EEPROM but doesn't return anything, I apparently don't know how to read from EEPROM
- read in EEPROM the speed of the main fan does not return anything either

All this did not make the strange behavior as seen in your video. But I haven't tested getEcoOff and getEcoTime yet, maybe this problem comes from there.

When I have time, I will analyze my official MicroNova WiFi module. In the meantime, you can use the program on the main branch. I made some changes but this version is fully functional.

</details>

<details>
  <summary><b>Français</b></summary>
  
Okay, merci.

Récapitulatif des tests effectués:
- la lecture en RAM de la valeur du thermostat retourne 0 (vous pouvez tester si vous voulez en décommentant `getTempSet();` et `delay(100);` juste avant)
- testé en lisant dans l'EEPROM mais cela ne routourne rien, je ne sais apparemment pas comment effectuer la lecture dans l'EEPROM
- la lecture en EEPROM de la vitesse du ventilateur principal ne retourne rien non plus du coup

Tout cela ne déclenchait pas l'étrange comportement du poêle comme vu dans votre vidéo mais je n'ai pas encore testé mes fonctions getEcoOff et getEcoTime, ce problème vient peut-être de là.

Lorsque j'aurai le temps, j'analyserais mon module WiFi officiel MicroNova. En attendant, vous pouvez utiliser le programme sur la branche main. J'y ai apporté quelques changements mais cette version est totalement fonctionnelle.

</details>

**Reply by bernardstef** [reply]
Alright, update to the latest version and no more issues with the stove turning on by itself. Thanks !

**Reply by philibertc** [reply]
Hello, on the serial debug off the easy connect module what are the logs when you turn the stove off?
I want to know if it sends 0x00 0xA1 or 0x06 0xA7

**Reply by bernardstef** [reply]
Hi !


Here is when I turn the stove On via the app :




[MQTT INFO] deliver_publish
[MQTT INFO] Data received: 270/270 bytes
I (107857060) [MQTT_CLIENT]: [APP] Publish topic: navel/94B97E7AA2B8
I (107857072) [MQTT_CLIENT]: [APP] Publish data[270/270 bytes]
I (107857073) [MQTT_CLIENT]: [REquest] data [{"mi":"9032f7d8-96a0-4981-8ea8-d09dad87eaa7","mt":"Q","r":{"ri":"","rp":0,"rj":"94B97E7AA2B8","rs":"XM"},"s":{"si":"","sp":0,"sj":"master/responses","ss":""},"md":10,"ts":"1633283294735","vu":"1633283295735","tz":"","dst":"","pl":{"BufferId":1,"cmd":"GetBufferReading"}}]
C (107857111) [JSON_router]: free heap mem = 4139680
I (107857113) [JSON_router]: Perform command GetBufferReading
I (107857144) [MQTT_CLIENT]: [Answer] data [{"mi":"9032f7d8-96a0-4981-8ea8-d09dad87eaa7","mt":"R","r":{"ri":"","rp":0,"rj":"master/responses","rs":""},"s":{"si":"","sp":0,"sj":"94B97E7AA2B8","ss":""},"md":10,"ts":"1633283294735","vu":"1633283295735","tz":"","dst":"","pl":{"BufferId":1,"cmd":"GetBufferReading","Items":[32847,32878,32857,32844,32884,32839,33,32828,32842,32845,32894,32860,33020,32826,33019,32867,232,32831,32853,32855,32850,33021,32882,32827,32835,32833,32863,6,32870,32851,51,32841,2,33017,32866,32881,33016,32877,32846,32836,32858,32795,32823,32880,32830,32838,32837,32861,32864,32798,32829,32843,32848,32872,32875,33018,32871,32874,1,32832,32852,32883,32895,32873,32876,32862,52,32859,32893,32840,32869,32868,32849,32834,32854,32856,32879,32825,32865],"Values":[0,0,0,0,0,0,0,1,0,0,2,0,16,96,3,102,32,1,0,0,0,33,0,135,0,1,0,129,0,0,0,0,0,25,96,0,7,0,0,51,0,30,0,0,1,0,52,0,0,1,1,0,0,0,0,72,0,0,37,1,0,0,2,0,0,0,0,0,22,0,0,0,0,1,0,0,0,0,0]}}]
[MQTT INFO] Sending...956 bytes
[MQTT INFO] Queuing publish, length: 956, queue size(0/8192)
C (107857231) [MQTT]: free heap mem = 4139396
[MQTT INFO] Read len 378/4096
[MQTT INFO] deliver_publish
[MQTT INFO] Data received: 355/355 bytes
I (107857983) [MQTT_CLIENT]: [APP] Publish topic: navel/94B97E7AA2B8
I (107857994) [MQTT_CLIENT]: [APP] Publish data[355/355 bytes]
I (107857995) [MQTT_CLIENT]: [REquest] data [{"mi":"5c28d820-7610-4629-90e2-7e1812a4dd48","mt":"C","r":{"ri":"","rp":0,"rj":"94B97E7AA2B8","rs":"XM"},"s":{"si":"","sp":0,"sj":"master/responses","ss":""},"md":10,"ts":"1633283295681","vu":"1633283296681","tz":"","dst":"","pl":{"BitData":[8],"Endianess":["L"],"Items":[232],"Masks":[65535],"Protocol":"RWMSmaster","Values":[85],"cmd":"RequestWriting"}}]
C (107858035) [JSON_router]: free heap mem = 4139156
I (107858036) [JSON_router]: Perform command RequestWriting
read for mask addr:00E8  value:0020
C (107858057) [BASE_COM]: write in RAM address: 00E8 value:85
write with mask addr:00E8 mask:FFFF value:0055
I (107858087) [MQTT_CLIENT]: [Answer] data [{"mi":"5c28d820-7610-4629-90e2-7e1812a4dd48","mt":"A","r":{"ri":"","rp":0,"rj":"master/responses","rs":""},"s":{"si":"","sp":0,"sj":"94B97E7AA2B8","ss":""},"md":10,"ts":"1633283295681","vu":"1633283296681","tz":"","dst":"","pl":{"Cmd":"RequestWriting"}}]
[MQTT INFO] Sending...293 bytes
[MQTT INFO] Queuing publish, length: 293, queue size(0/8192)
C (107858118) [MQTT]: free heap mem = 4139276


I think what's interesting are the "write" commands ?



Here is when I turn the stove off via the app :



[MQTT INFO] deliver_publish
[MQTT INFO] Data received: 356/356 bytes
I (108171080) [MQTT_CLIENT]: [APP] Publish topic: navel/94B97E7AA2B8
I (108171091) [MQTT_CLIENT]: [APP] Publish data[356/356 bytes]
I (108171092) [MQTT_CLIENT]: [REquest] data [{"mi":"56a9103e-c29e-40dd-b946-0acb8d4e07e2","mt":"C","r":{"ri":"","rp":0,"rj":"94B97E7AA2B8","rs":"XM"},"s":{"si":"","sp":0,"sj":"master/responses","ss":""},"md":10,"ts":"1633283608217","vu":"1633283609217","tz":"","dst":"","pl":{"BitData":[8],"Endianess":["L"],"Items":[232],"Masks":[65535],"Protocol":"RWMSmaster","Values":[170],"cmd":"RequestWriting"}}]
C (108171132) [JSON_router]: free heap mem = 4139156
I (108171134) [JSON_router]: Perform command RequestWriting
read for mask addr:00E8  value:0020
C (108171153) [BASE_COM]: write in RAM address: 00E8 value:170
write with mask addr:00E8 mask:FFFF value:00AA
I (108171180) [MQTT_CLIENT]: [Answer] data [{"mi":"56a9103e-c29e-40dd-b946-0acb8d4e07e2","mt":"A","r":{"ri":"","rp":0,"rj":"master/responses","rs":""},"s":{"si":"","sp":0,"sj":"94B97E7AA2B8","ss":""},"md":10,"ts":"1633283608217","vu":"1633283609217","tz":"","dst":"","pl":{"Cmd":"RequestWriting"}}]
[MQTT INFO] Sending...293 bytes
[MQTT INFO] Queuing publish, length: 293, queue size(0/8192)
C (108171213) [MQTT]: free heap mem = 4139276


Cheers !

**Reply by bernardstef** [reply]
Maybe interesting too : 

Set temp to 22°c:
C (108343748) [BASE_COM]: write in EEP address: 007D value:22

set flame power to  4
C (108453733) [BASE_COM]: write in EEP address: 007F value:4

set fan speed to 3
C (108480861) [BASE_COM]: write in EEP address: 007E value:3

Set eco stop to off (on is "1")
C (108514857) [BASE_COM]: write in EEP address: 001E value:0

set eco timeout to 25 minutes
C (108549261) [BASE_COM]: write in EEP address: 001B value:25

**Reply by philibertc** [reply]
Okay, thank you very much! Your module is a gold mine.

**Reply by bernardstef** [reply]
hehe, yes ! don't hesitate to ask :+1:

**Reply by philibertc** [reply]
Hello!
Can you try the program by replacing line 73:
```cpp
const char stoveOff[4] = {0x80, 0x21, 0x06, 0xA7};
```

by:
```cpp
const char stoveOff[4] = {0x80, 0x21, 0xAA, 0xCB};
```

Does it work when you send OFF via MQTT?
Thank you for your help!

**Reply by bernardstef** [reply]
Hi !
Unfortunately, it doesn't seem to react when I send OFF with this modification ..

**Reply by philibertc** [reply]
Oups, I made a mistake it's `const char stoveOff[4] = {0x80, 0xEA, 0xAA, 0xD4};`
For another person, it stays on "Final cleaning" indefinitely, so I was looking to replicate what the official module does.

![image](https://user-images.githubusercontent.com/57588282/135902407-25a18197-1c79-4520-960f-5bd016904c68.png)

**Reply by bernardstef** [reply]
Sorry, still no reaction from the stove ....
I did put the first command and it works back (as it did originally) : const char stoveOff[4] = {0x80, 0x21, 0x06, 0xA7};

**Reply by philibertc** [reply]
Okay, thanks for your tests.
So, with {0x80, 0x21, 0x06, 0xA7}, it does "Final cleaning" then "Off" correctly?

**Reply by bernardstef** [reply]
I just tried to make sure, and yes it does turn off after the final cleaning when I used the MQTT off command, 'with the syntax you described.
Although it takes some time to turn off, maybe 10 to 20 minutes.

**Reply by philibertc** [reply]
Hello, do you have got a logic analyser?

**Reply by bernardstef** [reply]
Hi,

Unfortunately, not ... I Have an oscilloscope, but it doesn't have this function..

**Reply by paciotti** [reply]
Hello
I read someone can play with the official interface by micronova, so I'll try to ask:
I have a Hydro stove that doesn't turn ON (need to double check the OFF command, probably going to OFF without cleaning).
Everything else seems to work correctly until I send the ON command, then I read wrong values from memory. Force OFF seems to work too.
I'm looking for some advice because I can't go on with blind trying...
Thanks!

P.S. I made an adapter with a RPi Zero and python. I will share my work once is done, if someone is interested.

**Reply by philibertc** [reply]
Hello,
Sorry for the late reply. Unfortunately, this problem seems difficult to solve. I will think about it but for now the only thing I can suggest is to add an infrared led to simulate the official remote control. The information for the IR protocol is [here](https://k3a.me/ir-controller-for-pellet-stove-with-micronova-controller-stufe-e-pellet-aria-ir-telecomando/).
